pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/ModularLoadManagerImpl.java

```diff
@@ -1131,11 +1161,16 @@ public void writeBundleDataOnZooKeeper() {
         // Write the bundle data to metadata store.
         List<CompletableFuture<Void>> futures = new ArrayList<>();
 
-        for (Map.Entry<String, BundleData> entry : loadData.getBundleData().entrySet()) {
-            final String bundle = entry.getKey();
-            final BundleData data = entry.getValue();
-            futures.add(
-                    pulsarResources.getLoadBalanceResources().getBundleDataResources().updateBundleData(bundle, data));
+        // use synchronized to protect bundleArr.
+        synchronized (this) {
+            int updateBundleCount = selectTopKBundle();
+            for (int i = 0; i < updateBundleCount; i++) {
+                final Map.Entry<String, BundleData> entry = (Map.Entry<String, BundleData>) bundleArr.get(i);
+                final String bundle = entry.getKey();
+                final BundleData data = entry.getValue();
+                futures.add(pulsarResources.getLoadBalanceResources().getBundleDataResources()
+                        .updateBundleData(bundle, data));
```

Comment:
```suggestion
            bundleArr.stream().limit(updateBundleCount).forEach(entry -> futures.add(
                    pulsarResources.getLoadBalanceResources().getBundleDataResources().updateBundleData(
                            entry.getKey(), (BundleData) entry.getValue())));
```

Using FP style could be more simple here to avoid explicit type cast between `Map.Entry<String, ? extends Comparable>` and `Map.Entry<String, BundleData>`. The code above just convert the value type.

Reply:
thanks for review, i push a new commit.